complex/ft_config.c

```diff
@@ -253,8 +253,17 @@ static int ft_parse_num(char *str, struct key_t *key, void *buf)
 		TEST_ENUM_SET_N_RETURN(str, FT_FUNC_SENDMSG, enum ft_class_function, buf);
 		TEST_ENUM_SET_N_RETURN(str, FT_FUNC_SENDV, enum ft_class_function, buf);
 		TEST_ENUM_SET_N_RETURN(str, FT_FUNC_SEND, enum ft_class_function, buf);
+		TEST_ENUM_SET_N_RETURN(str, FT_FUNC_INJECT_WRITEDATA, enum ft_class_function, buf);
+		TEST_ENUM_SET_N_RETURN(str, FT_FUNC_INJECT_WRITE, enum ft_class_function, buf);
```

Comment:
The string in the variable str is not NULL terminated. This is because the parser just returns the starting and ending positions (or length) of the element in the big string that contains the entire config. str points to the starting character of the element. So it has to be compared with the enum string using strncmp. Since one enum string (e.g. FT_FUNC_SEND) is substring of the other (e.g. FT_FUNC_SENDMSG), we have to first try and match with the longest string.

I think this can be fixed by duplicating the string and freeing it after parsing each element.